tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/analysis/IAnalysisModule.java

```diff
@@ -138,6 +139,34 @@ public interface IAnalysisModule extends ITmfComponent, IAnalysisRequirementProv
      */
     @Nullable Object getParameter(@NonNull String name);
 
+    /**
+     * @param configuration
+     *            A ITmfConfiguration
+     * @return true if it was successful or false if invalid config (TODO should we through exception?)
+     * @since 9.5
+     */
+    default boolean setConfiguration(ITmfConfiguration configuration) {
+        return false;
+    }
+
+    /**
+     * @param configuration
+     *            A ITmfConfiguration
+     * @return true if it was successful or false if invalid config (TODO should we through exception?)
+     * @since 9.5
+     */
+    default @Nullable ITmfConfiguration getConfiguration() {
```

Comment:
This use needed to be able to add the configuration IDataProviderDescriptors for derived data providers. This is not used by the analysis framework.